rllib_contrib/ddpg/src/rllib_ddpg/ddpg/ddpg.py

```diff
@@ -0,0 +1,313 @@
+import logging
+from typing import List, Optional, Type
+
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig, NotProvided
+from ray.rllib.algorithms.simple_q.simple_q import SimpleQ, SimpleQConfig
```

Comment:
what are we gonna do with these dependencies?

Reply:
My belief is that in order to write an implementation of ddpg in ray 2.5.1, one that reuses code as much as possible, you need to depend on ray 2.5.1 SimpleQ and SimpleQConfig.

Reply:
what if that algorithm itself moves to rllib_contrib

Reply:
We should combine such algorithms into one algorithm bucket I feel like. All of them have to be migrated together.

Reply:
I thought about this too e.g. grouping together all of the off policy algorithms.
In this bucket there would be simple_q, ddpg, apex_dqn, and apex_ddpg. 
If a user wants to make changes to 1 of these algorithms, then they might have to make changes to all 4 algorithms.

There's not one right answer here. Going with the current design means that if I add a feature to simpleq in rllib contrib then it won't be propagated to DDPG. Both solutions have their trade offs in terms of developer maintenance / burden.

What won me over is that rl developers like lower codebase complexity and they'll get that this way.

Reply:
so what is the maintenance burden if we go this way exactly? To minimize this burden and complexity we need to pin ALL dependency version, including the ones that depend on rllib_contrib itself. So we have to pip install -e on a specific commit / version.

Reply:
resolved offline -- We will write explicitly that all dependencies for an algorithm must be pinned explicitly.